pallets/collator-selection/src/lib.rs

```diff
@@ -306,13 +312,91 @@ pub mod pallet {
 				);
 			}
 
+			bounded_invulnerables.sort();
+
 			<Invulnerables<T>>::put(&bounded_invulnerables);
 			Self::deposit_event(Event::NewInvulnerables {
 				invulnerables: bounded_invulnerables.to_vec(),
 			});
 			Ok(().into())
 		}
 
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::add_invulnerable())]
+		pub fn add_invulnerable(
+			origin: OriginFor<T>,
+			new: <T>::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			
+			let invulnerables = Self::invulnerables().to_vec();
+
+			// ensure we don't overflow invulnerables quantity
+			let length = invulnerables.len();
+			ensure!((length as u32) < T::MaxInvulnerables::get(), Error::<T>::TooManyInvulnerables);
+			
+			// ensure this new invulnerable is not already nomitad
+			ensure!(!invulnerables.contains(&new), Error::<T>::AlreadyInvulnerable);
+			
+			// ensure this new invulnerable has registred a validator key
+			let validator_key = 
+				T::ValidatorIdOf::convert(new.clone()).ok_or(Error::<T>::NoAssociatedValidatorId)?;
+				ensure!(
+					T::ValidatorRegistration::is_registered(&validator_key),
+					Error::<T>::ValidatorNotRegistered
+				);
+			
+			<Invulnerables<T>>::try_mutate(|invulnerables| -> DispatchResult {
+				invulnerables.try_push(new.clone()).map_err(|_| Error::<T>::Unknown)?;
+				invulnerables.sort();
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::NewInvulnerable {
+				added: new,
+			});
+			Ok(().into())
+		}
+
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::remove_invulnerable())]
+		pub fn remove_invulnerable(
+		 	origin: OriginFor<T>,
+		 	to_remove: <T>::AccountId,
+		) -> DispatchResultWithPostInfo {
+			T::UpdateOrigin::ensure_origin(origin)?;
+			
+			let invulnerables = Self::invulnerables().to_vec();
+ 
+			// ensure invulnerable is actually in the list before trying to remove it
+			 ensure!(invulnerables.contains(&to_remove), Error::<T>::NotInvulnerable);
+
+		 	// check if old invulnerable still has associated validator keys before it is removed
+		 	let validator_key = 
+				T::ValidatorIdOf::convert(to_remove.clone()).ok_or(Error::<T>::NoAssociatedValidatorId)?;
+		 		ensure!(
+		 			!T::ValidatorRegistration::is_registered(&validator_key),
+		 			Error::<T>::ValidatorStillRegistered
+		 		);
+
+			// remove invulnerable from invulnerables list
+			<Invulnerables<T>>::try_mutate(|invulnerables| -> DispatchResult {
+				let pos = invulnerables
+					.binary_search(&to_remove)
+					.ok()
+					.ok_or(Error::<T>::NotInvulnerable)?;
+				invulnerables.remove(pos);
+				invulnerables.sort();
```

Comment:
You shouldn't need to sort on remove. If you remove an element from a sorted list it is still sorted.